test/Streamly/System/Process.hs

```diff
@@ -0,0 +1,205 @@
+module Main where
+
+import qualified Streamly.Internal.Prelude as S
+import qualified Streamly.System.Process as Proc
+import qualified Streamly.Internal.FileSystem.Handle as FH
+import qualified Streamly.Internal.Memory.ArrayStream as AS
+
+import System.IO (Handle, IOMode(..), openFile, hClose)
+import System.Process (proc, createProcess, waitForProcess)
+import System.Directory (removeFile)
+
+import Test.Hspec(hspec, describe)
+import Test.Hspec.QuickCheck
+import Test.QuickCheck 
+    ( forAll
+    , choose
+    , Property
+    , property
+    , listOf
+    , vectorOf
+    , counterexample
+    , Gen
+    )
+
+import Test.QuickCheck.Monadic (monadicIO, PropertyM, assert, monitor, run)
+import Control.Monad.IO.Class (MonadIO(..))
+import Control.Monad (when)
+
+import Data.List ((\\))
+import Data.Word (Word8)
+
+_a :: Word8
+_a = 97
+
+_z :: Word8
+_z = 122
+
+_A :: Word8
+_A = 65
+
+_Z :: Word8
+_Z = 90
+
+devRandom :: String
+devRandom = "/dev/random"
+
+devNull :: String
+devNull = "/dev/null"
+
+ddBinary :: String
+ddBinary = "/bin/dd"
+
+ddBlockSize :: Int
+ddBlockSize = 1024
+
+catBinary :: String
+catBinary = "/bin/cat"
+
+trBinary :: String
+trBinary = "/usr/bin/tr"
+
+minBlockCount :: Int
+minBlockCount = 1
+
+maxBlockCount :: Int
+maxBlockCount = 100
+
+minNumChar :: Int
+minNumChar = 1
+
+maxNumChar :: Int
+maxNumChar = 100 * 1024
+
+arrayChunkElem :: Int
+arrayChunkElem = 100
+
+toUpper :: Word8 -> Word8
+toUpper char =
+    if(_a <= char && char <= _z)
+    then (char - _a) + _A
+    else char
+
+listEquals :: (Show a, Eq a, MonadIO m)
+    => ([a] -> [a] -> Bool) -> [a] -> [a] -> PropertyM m ()
+listEquals eq process_output list = do
+    when (not $ process_output `eq` list) $ liftIO $ putStrLn $
+                  "process output " ++ show process_output
+             ++ "\nlist   " ++ show list
+             ++ "\nprocess output \\\\ list " ++ show (process_output \\ list)
+             ++ "\nlist \\\\ process output " ++ show (list \\ process_output)
+    when (not $ process_output `eq` list) $
+        monitor
+            (counterexample $
+                  "process output " ++ show process_output
+             ++ "\nlist   " ++ show list
+             ++ "\nprocess output \\\\ list " ++ show (process_output \\ list)
+             ++ "\nlist \\\\ process output " ++ show (list \\ process_output)
+             )
+    assert (process_output `eq` list)
+
+byteFile :: String
+byteFile = "./byteFile"
+
+charFile :: String
+charFile = "./largeCharFile"
+
+generateByteFile :: Int -> IO ()
+generateByteFile blockCount = 
+    let
```

Comment:
you can move this let inside do block. Also, combine the first line of `let` block with the `let` keyword. Apply the same to all code, in benchmarks as well.